src/Integration.Vsix.UnitTests/ErrorList/SonarErrorListDataSource_ClientIssueStoreTests.cs

```diff
@@ -0,0 +1,112 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration.Vsix;
+using SonarLint.VisualStudio.Integration.Vsix.ErrorList;
+using SonarLint.VisualStudio.IssueVisualization.Editor;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Selection;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.ErrorList
+{
+    [TestClass]
+    public class SonarErrorListDataSource_ClientIssueStoreTests
+    {
+        [TestMethod]
+        public void Get_ReturnsCorrectIssuesOneFactory()
+        {
+            var issue1 = Mock.Of<IAnalysisIssueVisualization>();
+            var issue2 = Mock.Of<IAnalysisIssueVisualization>();
+            var factory = SetUpFactoryWithIssues(issue1, issue2);
+
+            var testSubject = CreateTestSubject();
+
+            Get_ShouldContainIssues(testSubject);
+
+            testSubject.AddFactory(factory);
+            Get_ShouldContainIssues(testSubject, issue1, issue2);
+
+            testSubject.RemoveFactory(factory);
+            Get_ShouldContainIssues(testSubject);
+        }
+
+        [TestMethod]
+        public void Get_ReturnsCorrectIssuesTwoFactorîes()
+        {
+            var issue1 = Mock.Of<IAnalysisIssueVisualization>();
+            var issue2 = Mock.Of<IAnalysisIssueVisualization>();
+            var factory = SetUpFactoryWithIssues(issue1, issue2);
+
+            var issue3 = Mock.Of<IAnalysisIssueVisualization>();
+            var issue4 = Mock.Of<IAnalysisIssueVisualization>();
+            var factory2 = SetUpFactoryWithIssues(issue3, issue4);
+
+            var testSubject = CreateTestSubject();
+
+            Get_ShouldContainIssues(testSubject);
+
+            testSubject.AddFactory(factory);
+            Get_ShouldContainIssues(testSubject, issue1, issue2);
+
+            testSubject.AddFactory(factory2);
+            Get_ShouldContainIssues(testSubject, issue1, issue2, issue3, issue4);
+
+            testSubject.RemoveFactory(factory);
+            Get_ShouldContainIssues(testSubject, issue3, issue4);
+        }
+
+        private void Get_ShouldContainIssues(SonarErrorListDataSource testSubject, params IAnalysisIssueVisualization[] issues)
```

Comment:
I know we have some test classes with a syntax similar to this however, I couldn't remember any examples so I did it like this, please lmk if its oki

Reply:
Have a do-and-check method is ok.

Here, there are FluentAssertions [that compare collections](https://fluentassertions.com/collections/) are equivalent, so you could perhaps do the whole check in single line
e.g. `testSubject.Get().Should().Equal(issues)`